### ISSUE TITLE
isOldIE throws 'window is not defined' in non-browser environments

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [webpack-contrib/style-loader#272]
I'm using React server-side rendering in React.NET and compiling my app with `webpack`+`style-loader`+etc. `isOldIE` assumes `window` as a variable exists and can be falsy. The correct way to check for its existence would be `typeof` like in `lib/urls.js`.

```
ReferenceError: 'window' is undefined at Anonymous function (Script Document:700:1)
 at Anonymous function (Script Document:695:111)
 at module.exports (Script Document:702:23)
 at srcViewsRootStylesCss (Script Document:984:1)
 at __webpack_require__ (Script Document:50:9)
 ...
```

**Comment by indolent-developer** [comment]
Guys any update on this. do we have any PR for this. This should be simple one line change.

**Comment by evilebottnawi** [comment]
@michael-ciniawsky friendly ping

**Comment by michael-ciniawsky** [comment]
@indolent-developer Describe your use case please and provide `{Error}` logs and your `webpack.config.js`.

**Comment by indolent-developer** [comment]
@michael-ciniawsky i have reference this in library where we see issues. And regardless of that the code is not correct. Dont you agreee? The window checks are different in code else where.

**Comment by michael-ciniawsky** [comment]
@indolent-developer Then please provide more info about this issues :). Let's say I don't disagree but the checks seem to work and unless there isn't a confirmed bug within the scope of this loader there is less reason to change anything

**Comment by JoshuaKGoldberg** [comment]
tl;dr: The check does not work and that doesn't matter.

As per the OP, you can't check for existence of a global variable by seeing if it's truthy (which is what the check does). Just checking `if (window)` will throw an error. You have to use `if (typeof window !== "undefined")`. The same goes for `if (document)` verses `if (typeof document !== "undefined")`.

However, it's irrelevant whether this check is fixed, because style-loader relies on the existence of a `document` elsewhere in code. See [lib/addStyleUrl.js](https://github.com/webpack-contrib/style-loader/blob/master/lib/addStyleUrl.js). Merging in the linked PR still wouldn't get style-loader to work with server-side rendering (or more genera, execution in a non-browser environment).

michael-ciniawsky's comments in the PR are why this issue should be closed:
> style-loader requires a DOM and doesn't officially support SSR

For SSR specifically:
>  please use an proper alternative instead e.g isomorphic-style-loader. style-loader solely adds CSS via <style> to the DOM

**Comment by chardos** [comment]
Thanks for the replies guys.

What about the use case where a bundle includes both DOM dependent modules, and non-DOM dependent modules? I have a "component library" that has a bunch of react modules being compiled, but also a config file that needs to be accessed by DefinePlugin in another module. In my case, fixing the check would allow me to require just the single file I need.

Can we implement this fix?
